src/AzureAppConfigurationImpl.ts

```diff
@@ -433,31 +483,31 @@ export class AzureAppConfigurationImpl implements AzureAppConfiguration {
         }
 
         // check if any feature flag is changed
-        let needRefresh = false;
-        for (const selector of this.#featureFlagSelectors) {
-            const listOptions: ListConfigurationSettingsOptions = {
-                keyFilter: `${featureFlagPrefix}${selector.keyFilter}`,
-                labelFilter: selector.labelFilter,
-                pageEtags: selector.pageEtags
-            };
-            const pageIterator = listConfigurationSettingsWithTrace(
-                this.#requestTraceOptions,
-                this.#client,
-                listOptions
-            ).byPage();
-
-            for await (const page of pageIterator) {
-                if (page._response.status === 200) { // created or changed
-                    needRefresh = true;
-                    break;
+        const funcToExecute = async (client) => {
+            const needRefresh = false;
+            for (const selector of this.#featureFlagSelectors) {
+                const listOptions: ListConfigurationSettingsOptions = {

```

Comment:
Why do you specify type for `listOptions`? This is the only place you specify the type in the whole function. Just curious about it, why treat it so special? And I strongly recommend to specify type for all non-primitive types in all functions.